core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1554,12 +1554,21 @@ private CompiledQuery createTable(final ExecutionModel model, SqlExecutionContex
                     }
                     throw SqlException.$(name.position, "table already exists");
                 }
+                CharSequence volumePath = createTableModel.getVolumePath();
+                if (volumePath != null) {
+                    if (!ff.isDirOrSoftLinkDir(path.of(volumePath).$())) {
+                        throw CairoException.critical(0).put("not a valid path for volume [path=").put(path).put(']');
+                    }
+                    if (!configuration.isAllowedVolumePath(volumePath)) {
+                        throw CairoException.critical(0).put("volume path is not allowed [path=").put(path).put(']');
+                    }
+                }
                 try {
                     if (createTableModel.getQueryModel() == null) {
                         if (createTableModel.getLikeTableName() != null) {
                             copyTableReaderMetadataToCreateTableModel(executionContext, createTableModel);
                         }
-                        engine.createTableUnsafe(executionContext.getCairoSecurityContext(), mem, path, createTableModel);
+                        engine.createTableUnsafe(executionContext.getCairoSecurityContext(), mem, path, volumePath != null, createTableModel);
```

Comment:
I don't see value of `volumePath` to be used, other than tested for `null`. This `volume` related change looks untested, is it ready to go in this PR?

Reply:
there is a test to exemplify the use: `ServerMainForeignTableTest.testServerMainCreateTableInAllowedVolume`.

the parser understands `IN VOLUME path` in create table statements and creates the table in the specific volume, and a symlink from the root/tablename to volume/tablename. Only preconfigured paths are valid in these statements.

It is tested, there are tests for the parser, for the compiler, in addition to this.

The mechanism is to call `TableUtils.createTable(` with an additional flag to tell it that path has in fact the root path to use, otherwise it is overriden with configuration.getRoot().

It is ready to go in this PR.